reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -618,4 +691,286 @@ public void shutdown() {}
+                child.requestResponse(payload).subscribe(new LatencySubscriber<>(subscriber, this));
+        }
+
+        @Override
+        public Publisher<Payload> requestStream(Payload payload) {
+            return subscriber ->
+                child.requestStream(payload).subscribe(new CountingSubscriber<>(subscriber, this));
+        }
+
+        @Override
+        public Publisher<Payload> requestSubscription(Payload payload) {
+            return subscriber ->
+                child.requestSubscription(payload).subscribe(new CountingSubscriber<>(subscriber, this));
+        }
+
+        @Override
+        public Publisher<Void> fireAndForget(Payload payload) {
+            return subscriber ->
+                child.fireAndForget(payload).subscribe(new CountingSubscriber<>(subscriber, this));
+        }
+
+        @Override
+        public Publisher<Void> metadataPush(Payload payload) {
+            return subscriber ->
+                child.metadataPush(payload).subscribe(new CountingSubscriber<>(subscriber, this));
+        }
+
+        @Override
+        public Publisher<Payload> requestChannel(Publisher<Payload> payloads) {
+            return subscriber ->
+                child.requestChannel(payloads).subscribe(new CountingSubscriber<>(subscriber, this));
+        }
+
+        ReactiveSocketFactory<T> getFactory() {
+            return factory;
+        }
+
+        synchronized double getPredictedLatency() {
+            long now = Clock.now();
+            long elapsed = Math.max(now - stamp, 1L);
+
+            double weight;
+            double prediction = median.estimation();
+
+            if (prediction == 0.0) {
+                if (pending == 0) {
+                    weight = 0.0; // first request
+                } else {
+                    // subsequent requests while we don't have any history
+                    weight = STARTUP_PENALTY + pending;
+                }
+            } else if (pending == 0 && elapsed > inactivityFactor * interArrivalTime.value()) {
+                // if we did't see any data for a while, we decay the prediction by inserting
+                // artificial 0.0 into the median
+                median.insert(0.0);
+                weight = median.estimation();
+            } else {
+                double predicted = prediction * pending;
+                double instant = instantaneous(now);
+
+                if (predicted < instant) { // NB: (0.0 < 0.0) == false
+                    weight = instant / pending; // NB: pending never equal 0 here
+                } else {
+                    // we are under the predictions
+                    weight = prediction;
+                }
+            }
+
+            return weight;
+        }
+
+        int getPending() {
+            return pending;
+        }
+
+        private synchronized long instantaneous(long now) {
+            return duration + (now - stamp0) * pending;
+        }
+
+        private synchronized long incr() {
```

Comment:
If `incr()` and `decr()` are only used by the inner class `LatencySubscriber`, should we move them there?